test/Microsoft.Extensions.DataIngestion.Tests/MakdownChunker/MarkdownChunker.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.DataIngestion.Tests
+{
+    public sealed class MarkdownChunker : DocumentChunker
+    {
+        private readonly int _headerLevelToSplitOn;
+        private readonly bool _stripHeaders;
+
+
+        public MarkdownChunker(MarkdownHeaderLevel HeaderLevelToSplitOn = MarkdownHeaderLevel.Header3, bool StripHeaders = true)
```

Comment:
From the API design point of view, I think it would be just simpler if this method was accepting an `int`, with some simple validation and XML comment that says that only `1-6` values are allowed.